mesonbuild/mparser.py

```diff
@@ -180,10 +179,7 @@ def lex(self, subdir):
                             lineno += len(lines) - 1
                             line_start = mo.end() - len(lines[-1])
                     elif tid == 'number':
-                        value = int(match_text)
-                    elif tid == 'hexnumber':
-                        tid = 'number'
-                        value = int(match_text, base=16)
+                        value = int(match_text, base=0)
```

Comment:
Not sure if we care, but what makes me slightly concerned that we depend on python implementation here, e.g. if they add/change something meson syntax will be affected.

Reply:
Yeah I can see that...

In particular, Python 2 used to recognize `0123` as octal syntax, the `0o123` was introduced in Python 2.7 and Python 3 no longer recognizes the `0123` syntax, only the `0o` prefix.

But given Meson syntax is Python-inspired, I think it's probably fine to keep this as is.